table/metadata_internal_test.go

```diff
@@ -491,3 +491,118 @@ func TestV1WriteMetadataToV2(t *testing.T) {
 	assert.NotContains(t, rawData, "schema")
 	assert.NotContains(t, rawData, "partition-spec")
 }
+
+func TestNewMetadataWithExplicitV1Format(t *testing.T) {
```

Comment:
It would be nice to have a nested struct in the test to validate the pre-order traversal :)

Reply:
that's tested and validated in the `schema_test.go` file that specifically tests the AssignFreshIDs function :)